hypothesis-python/tests/ghostwriter/test_ghostwriter.py

```diff
@@ -425,3 +426,18 @@ def test_unrepr_identity_elem():
 )
 def test_get_imports_for_strategy(strategy, imports):
     assert ghostwriter._imports_for_strategy(strategy) == imports
+
+
+def test_obj_name():
+    # Module paths (strings including a "/") should raise a meaningful UsageError
+    with pytest.raises(click.exceptions.UsageError) as e:
+        cli.obj_name("module/range.py")
+    assert e.match(
+        "Remember that the ghostwriter should be passed the name of a module, not a path."
+    )
+    # File names of modules (strings ending in ".py") should raise a meaningful UsageError
+    with pytest.raises(click.exceptions.UsageError) as e:
+        cli.obj_name("range.py")
+    assert e.match(
+        "Remember that the ghostwriter should be passed the name of a module, not a file."
```

Comment:
In this case, I think we could even suggest the correct invocation, adding e.g. `f"\n\tTry:    hypothesis write {s[:-3]}"` to the end of the message.  (though I'd also use `myscript` to avoid name collisions with a builtin)

Reply:
Great idea. I've now implemented this for the case of "myscript.py", but didn't make any suggestions in the case of a passed path (e.g, "directory/subdirectory/module.py" or its Windows equivalent). Do you think that would be helpful?

Reply:
Hmm, let's leave that for now - with `myscript.py` there's a concrete and unambiguous suggestion; with a path it's much less likely that our suggestion would actually be importable.